app/src/main/java/com/quran/labs/androidquran/data/QuranDataProvider.java

```diff
@@ -66,15 +69,26 @@ private static UriMatcher buildUriMatcher() {
 
   @Override
   public boolean onCreate() {
-    mQuranSettings = QuranSettings.getInstance(getContext());
     return true;
   }
 
   @Override
   public Cursor query(@NonNull Uri uri, String[] projection, String selection,
       String[] selectionArgs, String sortOrder) {
+    Context context = getContext();
+    if (!didInject) {
+      Context appContext = context == null ? null : context.getApplicationContext();
+      if (appContext instanceof QuranApplication) {
+        ((QuranApplication) appContext).getApplicationComponent().inject(this);
```

Comment:
intentionally did this, because from the documentation for `ContentProvider`:

> You should defer nontrivial initialization (such as opening, upgrading, and scanning databases) until the content provider is used (via query(Uri, String[], String, String[], String), insert(Uri, ContentValues), etc).